andela/ah-backend-athena
# #162163264 Send verification link

**What does this PR do?**
This pull request delivers the feature of sending a verification email to a user upon successful registration.

**Description of the tasks to be completed**
When a user creates an account, an email has to be sent to the email address he used while signing up to verify that the email really exists in order to avoid having ghost users in our database. This means that a user cannot log in without first verifying their account.

**How can this be manually tested?**
- Click on the _View deployment_ button immediately below this `PR` description, copy the link and paste it in postman.
- Go to the endpoint of `api/users/` and signup
- Login immediately afterwards(`api/users/login`) with the data you used while signing up
_You get a message telling you to first verify your email before you can login_
- Go to your email inbox and check for an email from `Authors heaven`, click on the link and then you can be able to login.
**_Signup_**
![image](https://user-images.githubusercontent.com/39732510/49636613-27dec180-fa14-11e8-8e3f-0c9433ec77cd.png)

**_Immediate Login_**
![image](https://user-images.githubusercontent.com/39732510/49636642-404edc00-fa14-11e8-85ac-371a786eff69.png)

**_Verification link_**
![image](https://user-images.githubusercontent.com/39732510/49636842-ccf99a00-fa14-11e8-9a69-2bc6c0346f61.png)
![image](https://user-images.githubusercontent.com/39732510/49636847-d4b93e80-fa14-11e8-9a67-554cd3ed454d.png)

**_Successful login_**
![image](https://user-images.githubusercontent.com/39732510/49636875-e8fd3b80-fa14-11e8-9d4c-7e4cd83e4399.png)
